syft/create_sbom.go

```diff
@@ -0,0 +1,139 @@
+package syft
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/dustin/go-humanize"
+	"github.com/scylladb/go-set/strset"
+	"github.com/wagoodman/go-progress"
+
+	"github.com/anchore/syft/internal/bus"
+	"github.com/anchore/syft/internal/relationship"
+	"github.com/anchore/syft/internal/sbomsync"
+	"github.com/anchore/syft/internal/task"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/event/monitor"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+)
+
+func CreateSBOM(src source.Source, cfg CreateSBOMConfig) (*sbom.SBOM, error) {
```

Comment:
from a conversation with @willmurphyscode 
```suggestion
func CreateSBOM(src source.Source, cfg *CreateSBOMConfig) (*sbom.SBOM, error) {
```

maybe move to a pointer here? then folks can use `nil` and we'd call `DefaultCreateSBOMConfig` on their behalf. There may be good alternatives here.